comms/dht/src/network_discovery/discovering.rs

```diff
@@ -170,14 +170,24 @@ impl Discovering {
                     }),
             })
             .await?;
-
+        let mut counter = 0;
+        let mut peers_received = HashSet::new();
         while let Some(resp) = stream.next().await {
+            counter += 1;
+            if counter > self.params.num_peers_to_request {
+                error!(target: LOG_TARGET, "Remote peer sent more peers than we requested.");
+                return Err(NetworkDiscoveryError::TooManyPeersReceived);
+            }
             let GetPeersResponse { peer } = resp?;
 
             let peer = peer.ok_or_else(|| NetworkDiscoveryError::EmptyPeerMessageReceived)?;
-            let new_peer = peer
+            let new_peer: UnvalidatedPeerInfo = peer
                 .try_into()
                 .map_err(NetworkDiscoveryError::InvalidPeerDataReceived)?;
+            if !peers_received.insert(new_peer.public_key.clone()) {
+                error!(target: LOG_TARGET, "Remote peer sent duplicate peer.");
```

Comment:
nit: I think WARN is more appropriate, error implies a local error. Same for the error! above

Reply:
agreed